### ISSUE TITLE
OnFileActivated not handled

### ISSUE THREAD

**Issue by touseefbsb** [Microsoft/WindowsTemplateStudio#456]
I added some file types as file associations to my app and then I opened the app through those file type directly, I know I should be handeling onFileActivated in this case, but I am not because I am assuming it is already handled in OnActivated handler in the app.xaml.cs file? If so then how can I get the file/files through this event?

Current Behaviour : app is stuck on splash screen ( when activated through file type assosiation )

**Comment by ralarcon** [comment]
Hi,

You can check out our sample for extending the activation handlers which implements a file association: 
* Doc: https://github.com/Microsoft/WindowsTemplateStudio/blob/master/docs/activation.md
* Source code: https://github.com/Microsoft/WindowsTemplateStudio/tree/master/samples/activation. 

Hope this helps.

**Comment by touseefbsb** [comment]
yes this helps thnks

**Comment by crutkas** [comment]
@touseefbsb did you feel the documentation in the template wasn't enough to help out here?  is there something you wish we had done to make this better?

**Comment by touseefbsb** [comment]
No it's good but in my case i wanted to directly navigate to a different view with out shell page so i had to seprately override the onfileactivated event , i cant do tht by following your documentation bcz tht way app navigates to shell page , but in general case ur docs are fine

**Comment by crutkas** [comment]
well, that to me shows a gap.  you want to do something we don't explain how to do :)

**Comment by touseefbsb** [comment]
yes I guess you are right the gap exists, these type of handlers should be more generic : i.e, control should be given to developer to override the default behaviour of navigating to the shell page. because mostly onFileactivated is used to display something different rather than general homepage of the app

**Comment by mrlacey** [comment]
@touseefbsb I'm curious about you wanting to navigate to a page not within the shell page. Are you mixing pages within the nav panel and blank pages at top level? 
I'd expect that if using the NavPanel it'd be more common to navigate to a page within the Shell and pass that the activated information.

**Comment by touseefbsb** [comment]
I am navigating to a media page directly so i dnt want hamburger on tht page tht is why

**Comment by mrlacey** [comment]
@touseefbsb Aah, yes. That's a good reason. :)

**Comment by touseefbsb** [comment]
yeah I can collaprse and hode the pane but the hamburger toggle button still remains which isnt a ideal experience so that is why I opted to directly navigate to it instead of going through shell page :)

**Comment by crutkas** [comment]
This sounds like a navpane and expectations issue.  I don't know if we would have assumed you would break out of the navpane.  Then we need to explain how to break back in once you're done viewing.

**Comment by touseefbsb** [comment]
yes exactly that can be addressed

**Comment by crutkas** [comment]
@touseefbsb i created #482 for this topic, would this be something you'd want to try to update the docs with as sounds like you're currently walking through how to do this in your app?
